IBM/carbon-addons-iot-react
# table(fix): column width not affected by columns prop after init

Closes #1214

**Summary**
The column widths does not update when the table’s column prop is modified after the initial render. This bug has been “hidden” by the fact that the TableHead handles the column widths in its own state, so column resizing wa still working.

This issue is a fix, where the larger problem and suggested solution is defined here:
https://github.com/IBM/carbon-addons-iot-react/issues/1224

**Change List (commits, features, bugs, etc)**
- Updated a table story to feed back the `onColumnResize` data as modified `columns` prop
- Updated the TableHead to handle changes in the `columns` prop 
- Added a unit test that exposed the bug  

**Acceptance Test (how to verify the PR)**
- Verify thet the test is ok
- Verify that resizing is working properly for the table stories (Especially for for the updated story watson-iot-table--with-resize-oncolumnresize-callback-and-no-initial-column-width)
